test/contrib/dropbox_test.py

```diff
@@ -0,0 +1,368 @@
+DROPBOX_TEST_DIR_TO_DELETE = DROPBOX_TEST_PATH + '/dir_to_delete'
+DROPBOX_TEST_FILE_TO_DELETE_2 = DROPBOX_TEST_DIR_TO_DELETE + '/test3.2.txt'
+DROPBOX_TEST_FILE_TO_DELETE_1 = DROPBOX_TEST_DIR_TO_DELETE + '/test3.1.txt'
+DROPBOX_TEST_FILE_TO_COPY_ORIG = DROPBOX_TEST_PATH + '/dir4/test4.txt'
+DROPBOX_TEST_FILE_TO_MOVE_ORIG = DROPBOX_TEST_PATH + '/dir3/test3.txt'
+
+# All the following paths will be used by the tests
+DROPBOX_TEST_SMALL_FILE = DROPBOX_TEST_PATH + '/dir/small.txt'
+DROPBOX_TEST_LARGE_FILE = DROPBOX_TEST_PATH + '/dir/big.bin'
+
+DROPBOX_TEST_FILE_TO_COPY_DEST = DROPBOX_TEST_PATH + '/dir_four/test_four.txt'
+
+DROPBOX_TEST_FILE_TO_MOVE_DEST = DROPBOX_TEST_PATH + '/dir_three/test_three.txt'
+DROPBOX_TEST_OUTER_DIR_TO_CREATE = DROPBOX_TEST_PATH + '/new_folder'
+DROPBOX_TEST_DIR_TO_CREATE = DROPBOX_TEST_OUTER_DIR_TO_CREATE + '/inner_folder'
+
+DROPBOX_TEST_SIMPLE_DIR_TO_CREATE_AND_DELETE = DROPBOX_TEST_PATH + '/another_new_folder'
+
+DROPBOX_TEST_FILE_TO_UPLOAD_BZIP2 = DROPBOX_TEST_PATH + '/bin.file'
+DROPBOX_TEST_FILE_TO_UPLOAD_TEXT = DROPBOX_TEST_PATH + '/text.txt'
+DROPBOX_TEST_FILE_TO_UPLOAD_BIN = DROPBOX_TEST_PATH + '/file.bin'
+DROPBOX_TEST_FILE_TO_UPLOAD_LARGE = DROPBOX_TEST_PATH + '/file.blob'
+
+DROPBOX_TEST_NON_EXISTING_FILE = DROPBOX_TEST_SIMPLE_DIR + 'ajdlkajfal'
+
+
+@attr('dropbox')
+class TestClientDropbox(unittest.TestCase):
+    def setUp(self):
+        self.luigiconn = DropboxClient(DROPBOX_APP_TOKEN)
+
+        self.dropbox_api = dropbox.dropbox.Dropbox(DROPBOX_APP_TOKEN)
+        self.dropbox_api.files_upload(b'hello', DROPBOX_TEST_SIMPLE_FILE)
+        self.dropbox_api.files_upload(b'hello2', DROPBOX_TEST_FILE_IN_DIR)
+        self.dropbox_api.files_upload(b'hello3', DROPBOX_TEST_FILE_TO_MOVE_ORIG)
+        self.dropbox_api.files_upload(b'hello4', DROPBOX_TEST_FILE_TO_COPY_ORIG)
+        self.dropbox_api.files_upload(b'hello3.1', DROPBOX_TEST_FILE_TO_DELETE_1)
+        self.dropbox_api.files_upload(b'hello3.2', DROPBOX_TEST_FILE_TO_DELETE_2)
+
+    def tearDown(self):
+        self.dropbox_api.files_delete_v2(DROPBOX_TEST_PATH)
+        self.dropbox_api._session.close()
+
+    def test_exists(self):
+        self.assertTrue(self.luigiconn.exists('/'))
+        self.assertTrue(self.luigiconn.exists(DROPBOX_TEST_PATH))
+        self.assertTrue(self.luigiconn.exists(DROPBOX_TEST_SIMPLE_DIR))
+        self.assertTrue(self.luigiconn.exists(DROPBOX_TEST_SIMPLE_DIR + '/'))
+        self.assertTrue(self.luigiconn.exists(DROPBOX_TEST_SIMPLE_DIR + '//'))
+        self.assertTrue(self.luigiconn.exists(DROPBOX_TEST_SIMPLE_FILE))
+
+        self.assertFalse(self.luigiconn.exists(DROPBOX_TEST_SIMPLE_FILE + '/'))
+        self.assertFalse(self.luigiconn.exists(DROPBOX_TEST_SIMPLE_FILE + '//'))
+        self.assertFalse(self.luigiconn.exists(DROPBOX_TEST_NON_EXISTING_FILE))
+
+    def test_listdir_simple(self):
+        list_of_dirs = self.luigiconn.listdir(DROPBOX_TEST_PATH)
+        self.assertTrue('/' not in list_of_dirs)
```

Comment:
is this just to show that none of the files or subdirs in the dropbox_test_path include trailing slashes?

Reply:
This is to check that when we are listing directory '/foobar', we don't return the '/' directory. 

It was done to show that we are only listing the contents of a given path, and not the contents of the whole Dropbox account.